### ISSUE TITLE
Enable filtering of data before metrics are calculated

### ISSUE THREAD

**Issue by mofii** [Orchard-Ultrasound-Innovation/IntensityScans.jl#35]
When measuring the metrics we really should be filtering the data first.
This could be a simple bandpass filter or a matched filter.

For a bandpass filter this could be set directly with:
```Julia
set_filter(obj, f_low, f_high)
```

For a matched filter or different type of bandpass filer the filter kernel could be set by the user with something like:
If the user can 
```julia
set_filter_kernel(obj, kernel)
```

**Comment by mofii** [comment]
@quintinstedman @sanliergun @Gerard-OUS 
Please give us your thoughts on this as well

**Comment by quintinstedman** [comment]
Yes, I think we should have that capability. What are everyone's thoughts on doing it in Julia vs. Matlab? A lot of us use Matlab for signal processing, and it already has lots of filtering functions included with the signal processing toolbox. If we do it in Julia then we can see the metrics right away, but maybe it's a lower priority if we can already do it in Matlab.
